### PR TITLE
Remove TestCollectionUtils.setOf and use Java 11 Set.of() instead [HZ-3080]

### PR DESCRIPTION
**Pull request by orcunc**
- Java 11 provides the same functionality
- Also clean up unnecessary types to diamond operator in the changed tests

Jira : https://hazelcast.atlassian.net/browse/HZ-3080

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

**Review by JackPGreen — COMMENTED** [review]
@orcunc the returned HashSet used to be mutable, but Set.of is immutable.

If you want mutability, Guava’s Sets.newHashSet is a drop-in replacement.

If the tests pass I don’t think this is an issue, but it’s a change nevertheless.

Also 👍 for removing redundant type specification!

**Comment by orcunc** [comment]
> @orcunc the returned HashSet used to be mutable, but Set.of is immutable.
> 
> If you want mutability, Guava’s Sets.newHashSet is a drop-in replacement.
> 
> If the tests pass I don’t think this is an issue, but it’s a change nevertheless.
> 
> Also 👍 for removing redundant type specification!

Yes Set.of() returns Immutable set, but it does not affect the tests

**Review by JackPGreen — APPROVED** [review]

File: hazelcast/src/test/java/com/hazelcast/internal/networking/nio/iobalancer/MonkeyMigrationStrategyTest.java [hunk 1]
```diff
@@ -52,16 +51,16 @@ public class MonkeyMigrationStrategyTest extends HazelcastTestSupport {
 
     @Test
     public void imbalanceDetected_shouldReturnFalseWhenNoPipelineExist() {
-        ownerPipelines.put(imbalance.srcOwner, Collections.<MigratablePipeline>emptySet());
+        ownerPipelines.put(imbalance.srcOwner, Collections.emptySet());
```
Comment (arodionov) [inline hazelcast/src/test/java/com/hazelcast/internal/networking/nio/iobalancer/MonkeyMigrationStrategyTest.java:54]: Minor
Why not `Set.of()` either here?

**Review by arodionov — APPROVED** [review]

File: hazelcast/src/test/java/com/hazelcast/internal/networking/nio/iobalancer/MonkeyMigrationStrategyTest.java (hunk 1 above)
Comment (orcunc) [reply hazelcast/src/test/java/com/hazelcast/internal/networking/nio/iobalancer/MonkeyMigrationStrategyTest.java:54]: Hi. I just removed the type of the diamond operator. I did not want to change the original test.

**Pull request merged**